contracts/BoostedMasterChefJoe.sol

```diff
@@ -0,0 +1,621 @@
+            uint256 lpSupply = pool.lpToken.balanceOf(address(this));
+            if (lpSupply > 0) {
+                uint256 secondsElapsed = block.timestamp.sub(pool.lastRewardTimestamp);
+                uint256 joeReward = secondsElapsed.mul(joePerSec()).mul(pool.allocPoint).div(totalAllocPoint);
+                pool.accJoePerShare = pool.accJoePerShare.add(
+                    (joeReward.mul(ACC_TOKEN_PRECISION).div(pool.totalBoostedAmount))
+                );
+            }
+            pool.lastRewardTimestamp = block.timestamp;
+            poolInfo[pid] = pool;
+            emit UpdatePool(pid, pool.lastRewardTimestamp, lpSupply, pool.accJoePerShare);
+        }
+    }
+
+    /// @notice Deposit LP tokens to BMCJ for JOE allocation.
+    /// @param pid The index of the pool. See `poolInfo`.
+    /// @param amount LP token amount to deposit.
+    function deposit(uint256 pid, uint256 amount) external nonReentrant {
+        harvestFromMasterChef();
+        updatePool(pid);
+        PoolInfo storage pool = poolInfo[pid];
+        UserInfo storage user = userInfo[pid][msg.sender];
+
+        if (user.amount > 0) {
+            // Harvest JOE
+            uint256 boostedAmount = user.amount.mul(getUserBoost(pid, msg.sender)).div(1000);
+
+            uint256 pending = boostedAmount.mul(pool.accJoePerShare).div(ACC_TOKEN_PRECISION).sub(user.rewardDebt).add(
+                claimableJoe[pid][msg.sender]
+            );
+            claimableJoe[pid][msg.sender] = 0;
+            JOE.safeTransfer(msg.sender, pending);
+            emit Harvest(msg.sender, pid, pending);
+        }
+
+        uint256 balanceBefore = pool.lpToken.balanceOf(address(this));
+        pool.lpToken.safeTransferFrom(msg.sender, address(this), amount);
+        uint256 receivedAmount = pool.lpToken.balanceOf(address(this)).sub(balanceBefore);
+
+        // Effects
+
+        /// If a user already has an amount we need to update the `poolInfo`
+        /// `totalBoostedAmount` and `totalVeJoe` variables relative to their
+        /// old amount. Otherwise we can just add the new amounts onto the
+        /// existing.
+
+        // Update the `pool.totalVeJoe`.
+        pool.totalVeJoe = pool.totalVeJoe.sub(user.veJoeBalance).add(VEJOE.balanceOf(msg.sender));
+        user.veJoeBalance = VEJOE.balanceOf(msg.sender);
+
+        // Update the `pool.totalBoostedAmount`.
+        uint256 userBoost = getUserBoost(pid, msg.sender);
+        if (user.amount == 0) {
+            pool.totalBoostedAmount = pool.totalBoostedAmount.add(amount.mul(userBoost).div(1000));
+        } else {
+            pool.totalBoostedAmount = pool.totalBoostedAmount.sub(user.amount.mul(userBoost).div(1000)).add(
+                user.amount.add(amount).mul(userBoost).div(1000)
+            );
+        }
+
+        user.amount = user.amount.add(receivedAmount);
+        user.rewardDebt = user
+            .amount
+            .sub(amount)
+            .mul(getUserBoost(pid, msg.sender))
```

Comment:
can we reuse the `userBoost` on line 446 or is this called again intentionally here?

Reply:
:+1: this is fixed now